featuretools/primitives/standard/transform/binary/multiply_numeric_boolean.py

```diff
@@ -53,15 +54,12 @@ class MultiplyNumericBoolean(TransformPrimitive):
     def get_function(self):
         def multiply_numeric_boolean(ser1, ser2):
             if pdtypes.is_bool_dtype(ser1):
-                mask = ser1
+                bools = ser1
                 vals = ser2
             else:
-                mask = ser2
+                bools = ser2
                 vals = ser1
-            vals_not_null = vals.notnull()
-            # Only apply mask where the input is not null
-            mask = mask.where(vals_not_null)
-            result = vals.where(mask, mask.replace({False: 0}))
+            result = vals * bools.astype("Int64")
             # Replace all pd.NA with np.nan to avoid WW init error
             result = result.replace({pd.NA: np.nan})
```

Comment:
Do we need this line? In the doctest, the result has a `pd.NA`.

Reply:
That's a good question. On the surface it seems like it is no longer necessary, but let me look a bit closer. 

One thing I'm concerned about - the doc test doesn't actually initialize Woodwork, that happens later in the CFM process and this comment said it was done to avoid a WW init error. The doctest wouldn't cover that, and also the doctest example doesn't contain a `np.nan` because the return type is `Int64` which cannot hold `np.nan`, so the replace call does nothing in that case.

Reply:
I ran DFS/CFM using a dataframe that had all the column dtype combinations that we should run into, and it worked successfully. That leads me to believe this `replace` call is no longer required. I'll try to remove and see if our CI passes for the min package versions as well.

Here are the features that were generated and computed successfully:
```
[<Feature: int>,
 <Feature: Int_no_null>,
 <Feature: Int_null>,
 <Feature: float>,
 <Feature: float_null>,
 <Feature: bool>,
 <Feature: boolean_no_null>,
 <Feature: boolean_null>,
 <Feature: bool * Int_no_null>,
 <Feature: bool * Int_null>,
 <Feature: bool * float>,
 <Feature: bool * float_null>,
 <Feature: bool * int>,
 <Feature: boolean_no_null * Int_no_null>,
 <Feature: boolean_no_null * Int_null>,
 <Feature: boolean_no_null * float>,
 <Feature: boolean_no_null * float_null>,
 <Feature: boolean_no_null * int>,
 <Feature: boolean_null * Int_no_null>,
 <Feature: boolean_null * Int_null>,
 <Feature: boolean_null * float>,
 <Feature: boolean_null * float_null>,
 <Feature: boolean_null * int>]
```